src/zap.ts

```diff
@@ -0,0 +1,55 @@
+import { ChainId, isEthereum, isFantom } from "./chain";
+import { EthAddress, ZeroAddress } from "./helpers";
+import { Token, VaultMetadataOverrides } from "./types";
+
+export type ZapInWith = keyof Token["supported"];
+
+type ZapInDetails = { isZapInSupported: boolean; zapInWith?: ZapInWith };
+
+type ZapInArgs = {
+  chainId: ChainId;
+  token?: Partial<Token>;
+  vaultMetadata: VaultMetadataOverrides | null;
```

Comment:
should we be using `VaultMetadata` instead of `VaultMetadataOverrides` ? 

Maybe not for this PR, and `VaultMetadataOverrides` should be extending `VaultMetadata`

Reply:
So this should actually be the whole vault, refactored with https://github.com/yearn/yearn-sdk/pull/280/commits/b62b06bda5904632f7592a413039ee2e0a08a305 following our discussion